drivers/platform/aducm3029/i2c.c

```diff
@@ -206,7 +211,17 @@ int32_t i2c_write(struct i2c_desc *desc,
 		return SUCCESS;
 	}
 
-	trans->bRepeatStart = (stop_bit == 1) ? 0 : 1;
+	if (stop_bit == 0) {
+		prologue_size = bytes_number;
+		prologue_data = malloc(bytes_number);
```

Comment:
So, the change works on the assumption that the `write(stop_bit=0)` followed by a `read(stop_bit=1)`.
I think we should somehow enforce this workflow, nu sure how exactly. But if we don't enforce it, and the user performs a `write(stop_bit=0)` followed by a `write(stop_bit=0)`, you'll end up with a memory leak (that global pointer allocated again) and possibly other side effects.

I otherwhise agree with the change, also tried it in ideal conditions and behaves as expected.

Reply:
I have 2 proposals:
 - create a static function for write/read in order to support stop bit also for read  and not duplicate code.
    Regarding the  memory leak, if multiple calls with stop_bit = 0 are executed then, maybe realloc can be used or just overwrite the previous prolog freeing the pointer. Memory can be deallocated in i2c_close if no free was called
 - Create a i2c_readwrite function with prolog which can be used to read a register and return -EINVAL because aducm is not directly supporting stop bit.